src/view/components/AlphaPickerContainer.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FunctionComponent, useCallback, useEffect, useRef, useState } from 'react';
+import AlphaPicker from '../../components/alphaPicker/alphaPicker';
+import { IQuery } from './type';
+
+type AlphaPickerProps = {
+    query: IQuery;
+    reloadItems: () => void;
+};
+
+const AlphaPickerContainer: FunctionComponent<AlphaPickerProps> = ({ query, reloadItems }: AlphaPickerProps) => {
+    const [ alphaPicker, setAlphaPicker ] = useState<AlphaPicker>();
+    const element = useRef<HTMLDivElement>(null);
+
+    alphaPicker?.updateControls(query);
+
+    const onAlphaPickerChange = useCallback((e) => {
+        const newValue = (e as CustomEvent).detail.value;
+        if (newValue === '#') {
+            query.NameLessThan = 'A';
```

Comment:
changing a prop directly like this is not recommended and very prune to errors.

You then call reloadItems, but if someone forgets to call this, in a future update, is very prune to it and can be hard to find the cause if you don't know what you need to call for this param to "react". 

So my suggestion is using useState in the parent component where the query is defined. Then pass the setQuery to the children that needs to modify this state. then call setQuery with a callback and do your logic inside it and return a modified query (you need to create a new object inside that callback so react can detect a new object was changed).

Another approach is using useReducer. Same thing but organized in a different manner. (more centralized so to speak)

Reply:
https://github.com/jellyfin/jellyfin-web/pull/3792/commits/9df95d4fcbea0f59809e56ad4fa4b1c92afe0081

Reply:
kind of resolves the issue, but it's still the same, relying on sub states to trigger an update to the query.

You should try to explore useReducer or using "setQuery" (with useState) with a callback and put your logic inside the callback to change or not the query state.

I could do it, but i don't have the time right now.

Reply:
@vitorsemeano https://github.com/jellyfin/jellyfin-web/pull/3792/commits/b88f09c63d7b75b97ebc8c4539f98bc8ecb1d136.

for the rest of query state SelectView, Sort and Filter dialog must convert to react. current implementation it just warps the button

Reply:
@vitorsemeano updated [use single state](https://github.com/jellyfin/jellyfin-web/pull/3792/commits/1b07551f538be313e56865263fdddcecda759801)